Remove FetchContent dependencies and use system argparse.
Tests are disabled as they require Catch2 which would need network access.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,25 +13,9 @@
   set_target_properties(${target} PROPERTIES CXX_EXTENSIONS OFF)
 endfunction()

-Include(FetchContent)
+find_package(argparse REQUIRED)

-FetchContent_Declare(
-  Catch2
-  GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-  GIT_TAG        v3.4.0
-)
-FetchContent_MakeAvailable(Catch2)
-list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/extras)
-include(CTest)
-include(Catch)
-enable_testing()

-FetchContent_Declare(
-    argparse
-    GIT_REPOSITORY https://github.com/p-ranav/argparse.git
-)
-FetchContent_MakeAvailable(argparse)

 include_directories(include)
 add_subdirectory(src)
-add_subdirectory(tests)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,5 +3,5 @@

 add_executable(gen-envel gen-envel.cc)
 set_default_props(gen-envel)
-target_link_libraries(gen-envel envel argparse)
+target_link_libraries(gen-envel envel argparse::argparse)
